test/e2e/specs/site-editor/template-registration.spec.js

```diff
@@ -0,0 +1,356 @@
+			)
+		).toBeHidden();
+
+		// Verify the plugin-registered template doesn't appear in the Site Editor.
+		await admin.visitSiteEditor( {
+			postType: 'wp_template',
+		} );
+		await blockTemplateRegistrationUtils.searchForTemplate( 'Custom' );
+		await expect(
+			page.getByText( 'Custom Template (overridden by the theme)' )
+		).toBeHidden();
+		// Verify the theme template shows the theme name as the author.
+		await expect( page.getByText( 'AuthorEmptytheme' ) ).toBeVisible();
+	} );
+
+	test( 'templates can be deleted if the registered plugin is deactivated', async ( {
+		admin,
+		editor,
+		page,
+		requestUtils,
+		blockTemplateRegistrationUtils,
+	} ) => {
+		// Make an edit to the template.
+		await admin.visitSiteEditor( {
+			postType: 'wp_template',
+		} );
+		await blockTemplateRegistrationUtils.searchForTemplate(
+			'Plugin Template'
+		);
+		await page.getByText( 'Plugin Template' ).click();
+		await expect(
+			editor.canvas.getByText( 'This is a plugin-registered template.' )
+		).toBeVisible();
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+			attributes: { content: 'User-customized template' },
+		} );
+		await editor.saveSiteEditorEntities( {
+			isOnlyCurrentEntityDirty: true,
+		} );
+
+		// Deactivate plugin.
+		await requestUtils.deactivatePlugin(
+			'gutenberg-test-block-template-registration'
+		);
+
+		// Verify template can be deleted.
+		await admin.visitSiteEditor( {
+			postType: 'wp_template',
+		} );
+		const deletedNotice = page
+			.getByLabel( 'Dismiss this notice' )
+			.getByText( `"Plugin Template" deleted.` );
+		const savedButton = page.getByRole( 'button', {
+			name: 'Saved',
+		} );
+		await blockTemplateRegistrationUtils.searchForTemplate(
+			'Plugin Template'
+		);
+		const searchResults = page.getByLabel( 'Actions' );
+		await searchResults.first().click();
+		await page.getByRole( 'menuitem', { name: 'Delete' } ).click();
+		await page.getByRole( 'button', { name: 'Delete' } ).click();
+
+		await expect( deletedNotice ).toBeVisible();
+		await expect( savedButton ).toBeVisible();
+
+		// Expect template to no longer appear in the Site Editor.
+		await expect( page.getByLabel( 'Actions' ) ).toBeHidden();
+
+		// Reactivate plugin.
+		await requestUtils.activatePlugin(
+			'gutenberg-test-block-template-registration'
+		);
+	} );
+
+	test( 'registered templates can be unregistered', async ( {
+		admin,
+		page,
+		blockTemplateRegistrationUtils,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postType: 'wp_template',
+		} );
+		await blockTemplateRegistrationUtils.searchForTemplate(
+			'Plugin Unregistered Template'
+		);
+		await expect(
+			page.getByText( 'Plugin Unregistered Template' )
+		).toBeHidden();
+	} );
+
+	test( 'WP default templates can be overridden by plugins', async ( {
+		page,
+	} ) => {
+		await page.goto( '?page_id=2' );
+		await expect(
+			page.getByText( 'This is a plugin-registered page template.' )
+		).toBeVisible();
+	} );
```

Comment:
This new test is failing on my freshly rebased branch. Any idea why?  #64479

Reply:
I left a comment on https://github.com/WordPress/gutenberg/pull/64479#issuecomment-2288819169.

Reply:
I think this line `await page.goto( '?page_id=2' );` might cause the issue. The test assumes that a page with ID = 2 will exist when it's run. The test that ran before in the same shard can remove all pages. Usually, the test file is responsible for setting up all required data.

The same probably applies to the `await page.goto( '/?cat=1' );` calls above.

Related discussion about DB snapshots: https://github.com/WordPress/gutenberg/issues/61719